bundles/framework/hierarchical-layerlist/Flyout.js

```diff
@@ -0,0 +1,749 @@
+/**
+ * @class Oskari.framework.bundle.hierarchical-layerlist.Flyout
+ *
+ * Renders the "all layers" flyout.
+ */
+Oskari.clazz.define('Oskari.framework.bundle.hierarchical-layerlist.Flyout',
+
+    /**
+     * @method create called automatically on construction
+     * @static
+     * @param {Oskari.framework.bundle.hierarchical-layerlist.HierarchicalLayerlistBundleInstance} instance
+     *    reference to component that created the flyout
+     */
+
+    function(instance) {
+        var me = this;
+        this.instance = instance;
+        this.service = this.instance.layerlistExtenderService;
+        this.container = null;
+        this.template = null;
+        this.state = null;
+        this.layerTabs = [];
+        this.filterTemplate = jQuery('<div class="filter filter-border"><center><div class="filter-icon"></div><div class="filter-text"></div></center></div>');
+
+        this._filterNewestCount = 20;
+        this._currentFilter = null;
+        this.mapLayerService = Oskari.getSandbox().getService('Oskari.mapframework.service.MapLayerService');
+        this.layerlistService = Oskari.getSandbox().getService('Oskari.mapframework.service.LayerlistService');
+
+        this.addedButtons = {};
+
+        this.layerlistService.on('Layerlist.Filter.Button.Add', function(button) {
+            me.addFilterTool(button.properties.text, button.properties.tooltip, button.properties.cls.active, button.properties.cls.deactive, button.filterId);
+        });
+        this._bindExtenderServiceListeners();
+    }, {
+
+        /*******************************************************************************************************************************
+        /* PRIVATE METHODS
+        *******************************************************************************************************************************/
+        /**
+         * Bind extender service listeners
+         * @method  _bindExtenderServiceListeners
+         * @private
+         */
+        _bindExtenderServiceListeners: function() {
+            var me = this;
+
+            me.service.on('option.added', function() {
+                me.populateLayers();
+            });
+
+            me.service.on('admin.changed', function() {
+                me.populateLayers();
+            });
+
+            me.service.on('admin-layer', function(data) {
```

Comment:
You should use the same naming scheme instead of some events having words separated with dots while others with dash.